raphtory/src/core/entities/graph/tgraph.rs

```diff
@@ -139,21 +139,28 @@ impl TemporalGraph {
     }
 
     pub(crate) fn layer_ids(&self, key: Layer) -> Result<LayerIds, GraphError> {
+        let valid_layers = self
```

Comment:
Seeing as we only need this in the error state - we should move this inside of the None/Ok or else - otherwise every other time we look up an edge we are gonna run it